### PR TITLE
[ONNX][bench] Deepcopy model to another device before export to avoid OOM

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118710

Prior to onnx export, the model is deepcopied to avoid modifications that may affect later performance profiling. However this increases the memory requirement on the device. 
This PR modifies the script to deepcopy and export the model on another device when possible.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -1222,6 +1222,31 @@ def __init__(
             self.model_dir / f"{model_name}_{self._COMPILER_NAME}.onnx"
         )
 
+    def _determine_deepcopy_target_device(self):
+        if current_device == "cpu":
+            target_device = "cpu"
+        else:
+            if torch.cuda.device_count() > 1:
+                # Copy to another cuda device to avoid OOM.
+                target_device = "cuda:1"
```
Comment (thiagocrepaldi) [inline benchmarks/dynamo/common.py:1231]: what if the user does CUDA_VISIBLE_DEVICE="2,3"?

By hardcoding a random device ID, things can go south. Maybe try to read the env var to see which devices are available. or use torch.cuda to get this list

**Review by thiagocrepaldi — APPROVED** [review]
Approved, but we could improve the ":1" hard-coding

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (BowenBao) [reply benchmarks/dynamo/common.py:1231]: it doesn't work that way. For `export CUDA_VISIBLE_DEVICE="2,3"` case, the available device id is still "0, 1" from code perspective.

**Pull request closed without merging**